tests/std/tests/P0355R7_calendars_and_time_zones_dates/test.cpp

```diff
@@ -0,0 +1,594 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <chrono>
+#include <type_traits>
+
+using namespace std;
+using namespace std::chrono;
+
+constexpr int y_min = -32767;
+constexpr int y_max = 32767;
+
+constexpr void day_test() {
+    day d{0u};
+    assert(static_cast<unsigned>(d) == 0u);
+    assert(d == 0d);
+    assert(++d == 1d);
+    assert(d++ == 1d);
+    assert(d == 2d);
+
+    assert(--d == 1d);
+    assert(d-- == 1d);
+    assert(d == 0d);
+
+    d += days{2};
+    assert(d == 2d);
+    d -= days{2};
+    assert(d == 0d);
+
+    assert(d < 2d);
+    assert(2d > d);
+
+    day d2{0u};
+    assert(!d2.ok());
+    ++d2;
+    for (int i = 1; i <= 31; ++i, ++d2) {
+        assert(d2.ok());
+    }
+    assert(!d2.ok());
+
+    assert(10d - 2d == days{8});
+}
+
+constexpr void month_test() {
+
+    month m{1u};
+    assert(static_cast<unsigned>(m) == 1u);
+    assert(m == January);
+    assert(++m == February);
+    assert(m++ == February);
+    assert(m == March);
+
+    assert(--m == February);
+    assert(m-- == February);
+    assert(m == January);
+
+    m += months{2};
+    assert(m == March);
+    m -= months{2};
+    assert(m == January);
+
+    for (unsigned i = 0; i <= 255; ++i) {

```

Comment:
That is a very good question I'd like to know the answer to myself.

(I assume it's because the idea of testing the full range came to me when doing months and forgot to apply it to days?)